### DISCUSSION TITLE
Discrepancy between write and read

### DISCUSSION THREAD

**Discussion by romansndlr** [spatie/laravel-data#233] [Q&A]
I have a use case where I have a typical "upsert" form. Some of the inputs are multi selects so for example in my DTO I'll have a "users" property which is a list of ids. So this works fine for my update and store actions... The problem is with the edit action, when I populate the form with an existing entity I need to give it not only a list of ids, but also a label for each id (the user's name for example). Should I just make the DTO accept for the users property both a list of ids and a list of label,id pairs?

**Comment by rubenvanassche** [comment]
You could probably create a DTO with an optional property of label. This property is then not needed in some cases.
